mantidimaging/gui/test/gui_system_spectrum_test.py

```diff
@@ -56,3 +57,71 @@ def test_add_roi(self) -> None:
             self.assertEqual(final_roi_count, initial_roi_count + 1)
             self.assertIn(f'roi_{i}', self.spectrum_window.roi_table_model.roi_names())
             self.assertIn(f'roi_{i}', self.spectrum_window.spectrum_widget.roi_dict)
+
+    def test_remove_roi(self) -> None:
+        QTest.mouseClick(self.spectrum_window.addBtn, Qt.MouseButton.LeftButton)
+        QTest.qWait(SHORT_DELAY)
+        initial_roi_count = self.spectrum_window.roi_table_model.rowCount()
+
+        QTest.mouseClick(self.spectrum_window.removeBtn, Qt.MouseButton.LeftButton)
+        QTest.qWait(SHORT_DELAY)
+        final_roi_count = self.spectrum_window.roi_table_model.rowCount()
+
+        self.assertEqual(final_roi_count, initial_roi_count - 1)
+        self.assertNotIn(f'roi_{initial_roi_count - 1}', self.spectrum_window.roi_table_model.roi_names())
+        self.assertNotIn(f'roi_{initial_roi_count - 1}', self.spectrum_window.spectrum_widget.roi_dict)
+
+    def test_change_roi_color(self) -> None:
+        QTest.mouseClick(self.spectrum_window.addBtn, Qt.MouseButton.LeftButton)
+        QTest.qWait(SHORT_DELAY)
+
+        new_color = (255, 0, 0, 255)
+        self.spectrum_window.spectrum_widget.change_roi_colour('roi_1', new_color)
+
+        self.assertEqual(self.spectrum_window.spectrum_widget.roi_dict['roi_1'].colour, new_color)
+
+    def test_rename_roi(self) -> None:
+        QTest.mouseClick(self.spectrum_window.addBtn, Qt.MouseButton.LeftButton)
+        QTest.qWait(SHORT_DELAY)
+
+        old_name = 'roi_1'
+        new_name = 'roi_renamed'
+        self.spectrum_window.presenter.rename_roi(old_name, new_name)
+
+        self.assertNotIn(old_name, self.spectrum_window.spectrum_widget.roi_dict)
+        self.assertIn(new_name, self.spectrum_window.spectrum_widget.roi_dict)
```

Comment:
Would be good here to step back and trigger it by modifying the table. It might work to use
`self.spectrum_window.presenter.roi_table_model.set_element()` to change the name, and then call `self.spectrum_window.presenter.roi_table_model.dataChanged.emit()`, or something similar.